implementation.md

```diff
@@ -0,0 +1,164 @@
+# Implementation Notes
+
+HyperApp's design is based on the [Elm Architecture](https://guide.elm-lang.org/architecture/) which enables you to create scalable browser-based applications using a functional paradigm.
+
+The logic of every program using this architecture will break into three parts :
+
+* **Model** - It represents the state of your application
+* **Update** - A way to update your state (actions)
+* **View** - View your state as HTML
+
+This pattern enforces you to write the code which is declarative and pure (immutable).
+
+## Virtual DOM
+HyperApp uses virtual DOM which provides a declarative way of representing the UI of your app. A virtual DOM tree is basically a data structure that consists of nested `vnodes` which describes a DOM tree. Whenever the state of your application changes, a virtual DOM tree is created to update the view. Then it ***diffs*** a `vnode` tree against its previous version and only modifies the DOM where there are changes.
+
+Read more about virtual DOM [here](https://github.com/Matt-Esch/virtual-dom).
+
+
+## Overview
+
+### Creating a `vnode` tree (hyperscript)
+
+#### `h(tag, properties, children)`
+
+`h()` takes a tag, an optional properties objects and an optional array of children or a single child that is a string. If you pass it an array of children, it will have child nodes.
+
+You can pass deeply nested tags and also components like this
+
+```javascript
+const Component = (data, children) => h("div", data, children)
+
+h(Component, { id: "foo" }, [h(Component, { id: "bar" })])
+```
+
+it will then create a `vnode` which consists of a `tag`, `props` and `children`.
+
+If you pass a number as a child, `h()` will parse it to string.
+
+### Creating an element
+
+#### `createElementFrom(node, isSVG)`
+
+It takes a `vnode` object and an optional `isSVG` option if the node is `svg`.
+
+It the type of a node is string, it will create a text node. Example -
+```javascript
+$ createElementFrom('Hello World!')
+'Hello World!'
+```
+
+If you provide a `vnode` object which includes a `tag` like this
+```javascript
+vnode = {
+  tag: 'div',
+  data: { id: 'foo' },
+  children: ["Hello World"]
+}
+```
+
+it would then create an element with that tag. Example -
+```javascript
+$ createElementFrom(h('div', { id: 'foo', style: { color: 'red'} }, ["bar"] ))
+<div id="foo" style="color: red;">bar</div>
+```
+
+More Examples -
+
+**Creating a `svg`**
+
+```javascript
+$ createElementFrom(h('svg', { width: "100", height: "100" }, h('circle', { cx: '50', cy: '50'})))
+
+<svg width="100" height="100">
```

Comment:
No, the markdown is wrong. 

<img width="611" alt="screen shot 2017-06-30 at 14 11 20" src="https://user-images.githubusercontent.com/56996/27721814-1c9aaac8-5d9e-11e7-9d20-27b2499451b7.png">

Also, what is `$`?

One more thing, don't use ```javascript. Use ```js or ```jsx.